raft/leaderElection/leaderelection.go

```diff
@@ -11,11 +11,23 @@ import (
 
 // LeaderElection implements Raft leader election
 func LeaderElection(config types.Configuration, IP string) {
-	// for !findALeader(config, IP) {
-	findALeader(config, IP)
-	log.Println("No leader found, repeating election")
-	// }
-	log.Println("Leader found!")
+	incrementTermAndBecomeCandidate(IP)
+	log.Printf("%s incremented term to %d\n", IP, types.ServerData[IP].CurrentTerm)
+
+	rand.Seed(time.Now().UnixNano())
+	timer := getTimer(150, 300)
+	log.Printf("%s set timeout at %d\n", IP, timeOut)
+
+	voteChan := make(chan int)
+	requestvotes.RequestVotes(config, IP, timer, voteChan)
+	votes := <-voteChan
```

Comment:
Instead of sending a channel and getting value back, we can make RequestVotes return votes directly. Thoughts?

Reply:
Sure. Works.